### PR TITLE
feat(docs-utils,dgeni): generate breadcrumbs from doc path

### LINKED ISSUE #2973: [FEAT] Breadcrumbs on top of docs pages
# :bulb: Feature request

## Feature Name
Breadcrumbs on top of docs pages

## The Desired Behavior
There should be breadcrumbs on the top of all docs pages so that a user knows where they are.

## Environment

<pre><code>
Daffodil version: X.Y.Z
Angular version: X.Y.Z 

 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

Dependent Tasks:
- [x] #3046
- [x] #3021

### LINKED ISSUE #3046: Generate breadcrumb data with dgeni

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: #3046, fixes: #2973

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
The guides and packages aren't quite as clean as the API since there isn't a doc at every level of the tree.

### CODE REVIEW

**Comment by xelaint** [comment]
@griest024 can you add the implementation of it using the `DaffBreadcrumbComponent`?

**Comment by griest024** [comment]
blocked by #3063

**Comment by griest024** [comment]
> @griest024 can you add the implementation of it using the `DaffBreadcrumbComponent`?

done

**Comment by griest024** [comment]
truncate current breadcrumb label

**Comment by griest024** [comment]
@xelaint ready for re-review

File: apps/daffio/src/app/docs/components/doc-viewer/doc-viewer.module.ts
Comment (xelaint) [inline apps/daffio/src/app/docs/components/doc-viewer/doc-viewer.module.ts]: I don't think it works like this? It's causing errors when I try to serve daffio via `npx ng s daffio -c dev-libs`

```
 Value could not be determined statically. [plugin angular-compiler]

    apps/daffio/src/app/docs/components/doc-viewer/doc-viewer.module.ts:23:11:
      23 │   imports: [
         ╵            ^

  Unable to evaluate this expression statically.

    apps/daffio/src/app/docs/components/doc-viewer/doc-viewer.module.ts:31:7:
      31 │     ...DAFF_BREADCRUMB_COMPONENTS,
         ╵        ~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by xelaint — CHANGES_REQUESTED** [review]

File: apps/daffio/src/app/docs/components/doc-viewer/doc-viewer.module.ts
Comment (griest024) [reply apps/daffio/src/app/docs/components/doc-viewer/doc-viewer.module.ts]: hm I'm not sure that we can import components like this. It works with the regular build but not `libs`. This needs to be investigated further before we merge all the standalone stuff

**Review by xelaint — DISMISSED** [review]

**Pull request merged**
